go-controller/pkg/ovn/pods.go

```diff
@@ -0,0 +1,118 @@
+package ovn
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"os/exec"
+	"strings"
+	"time"
+	"unicode"
+
+	kapi "k8s.io/client-go/pkg/api/v1"
+)
+
+func (oc *OvnController) getGatewayFromSwitch(logical_switch string) (string, string, error) {
+	var gateway_ip_mask_str string
+	var ok bool
+	if gateway_ip_mask_str, ok = oc.gatewayCache[logical_switch]; !ok {
+		gateway_ip_bytes, err := exec.Command(OVN_NBCTL, "--if-exists", "get",
+			"logical_switch", logical_switch,
+			"external_ids:gateway_ip").Output()
+		if err != nil {
+			glog.V(4).Infof("Gateway IP:  %s, %v", string(gateway_ip_bytes), err)
+			return "", "", err
+		}
+		gateway_ip_mask_str = strings.TrimFunc(string(gateway_ip_bytes), unicode.IsSpace)
+		gateway_ip_mask_str = strings.Trim(gateway_ip_mask_str, `"`)
+		oc.gatewayCache[logical_switch] = gateway_ip_mask_str
+	}
+	gateway_ip_mask := strings.Split(gateway_ip_mask_str, "/")
+	gateway_ip := gateway_ip_mask[0]
+	mask := gateway_ip_mask[1]
+	glog.V(4).Infof("Gateway IP: %s, Mask: %s", gateway_ip, mask)
+	return gateway_ip, mask, nil
+}
+
+func (oc *OvnController) deleteLogicalPort(pod *kapi.Pod) {
+	glog.V(4).Infof("Deleting pod: %s", pod.Name)
+	out, err := exec.Command(OVN_NBCTL, "lsp-del", fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Error in deleting pod network switch - %v(%v)", out, err)
+	}
+	return
+}
+
+func (oc *OvnController) addLogicalPort(pod *kapi.Pod) {
+
+	count := 30
+	logical_switch := pod.Spec.NodeName
+	for count > 0 {
+		if logical_switch != "" {
+			break
+		}
+		if count != 30 {
+			time.Sleep(1 * time.Second)
+			count--
+		}
+		p, err := oc.Kube.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			glog.Errorf("Could not get pod %s/%s for obtaining the logical switch it belongs to", pod.Namespace, pod.Name)
+			return
```

Comment:
Fair point. Added continue here.
The loop is primarily there because we are racing with the scheduler. If this controller catches the birth of a pod sooner than the scheduler, we want to give enough tries so that scheduler can assign the NodeName field for us (which decides the logical switch). Will matter in large clusters.